### PR TITLE
[WIP] New authentication forms

### PR DESCRIPTION
**Pull request by broskoski**
TODO:

- [x] Make login and signup actually work
- [x] Figure out integration with passport and current setup
- [x] Determine if we actually want a headline (i.e. "Are.na is a platform for thinking together") on any forms.
- [x] Routing
- [ ] Adapt signup form for accepting an invitation (this needs a corresponding query and mutation on the API as well)
- [ ] Reset password form
- [ ] Fix up input error styling
- [ ] Confirm that split-out confirmation app is still working as expected

---
Screenshots:
screenshot 2018-07-21 15 51 23
screenshot 2018-07-21 15 51 29
screenshot 2018-07-21 15 51 34

### CODE REVIEW

File: apps/authentication/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  apolloMiddleware,
+];
+
+const app = express();
+
+app.set('views', `${__dirname}/templates`);
+app.set('view engine', 'jade');
+
+app
+  .get(/^\/(sign_up|log_in|forgot)/, ...middlewareStack, (req, res) => req.apollo.render(StaticRoutes).then(apollo => res.render('index', { apollo })));
```
Comment (dzucconi) [inline apps/authentication/index.js]: interesting location for a line break :) some more might make this a bit more legible.

File: assets/auth.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { mount } from 'apps/authentication/client';
+
+$(mount);
```
Comment (dzucconi) [inline assets/auth.js]: a simple thing, but where possible let's get rid of jquery:

```javascript
document.addEventListener('DOMContentLoaded', mount);
```

File: react/components/AuthFormContainer/index.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+const AuthInput = styled(Input)`
+  margin: 1.5em 0;
+`;
+
+const Button = styled(GenericButton).attrs({ f: 2, type: 'submit' })`
+  margin: 0 auto;
+  padding-left: 2em;
+  padding-right: 2em;
+  color: ${x => x.theme.colors.gray.base};
```
Comment (dzucconi) [inline react/components/AuthFormContainer/index.js]: You can set all this stuff using the system attrs

File: assets/auth.js (hunk 2 above)
Comment (broskoski) [reply assets/auth.js]: 👍

File: react/components/AuthFormContainer/index.js (hunk 3 above)
Comment (broskoski) [reply react/components/AuthFormContainer/index.js]: Hm, how do you mean?

Comment (dzucconi) [reply react/components/AuthFormContainer/index.js]:
```
color: 'gray.base',
p: 6,
```

etc...

Don't make up your own values for things; use the scales we have: https://dzucconi.github.io/ervell/?selectedKind=Styles&selectedStory=Space&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

Comment (dzucconi) [reply react/components/AuthFormContainer/index.js]: https://dzucconi.github.io/ervell/?selectedKind=Button&selectedStory=GenericButton%20-%20colors&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

Comment (broskoski) [reply react/components/AuthFormContainer/index.js]: Ah you're talking about the padding values, not color, right?

Comment (dzucconi) [reply react/components/AuthFormContainer/index.js]: copy paste that snippet into your attrs declaration

Comment (dzucconi) [reply react/components/AuthFormContainer/index.js]: You don't need a function to set the color here.

Comment (dzucconi) [reply react/components/AuthFormContainer/index.js]: it'd probably be helpful to publish your storybook

**Comment by broskoski** [comment]
First pass at doing per-field messages. Definitely some nuance to consider here.

---
screen shot 2018-07-25 at 7 04 16 pm

**Comment by dzucconi** [comment]
Where's the forms issue in Basecamp? It's ambiguous to me as to which notice applies to which input; primarily because of the uniform margins.

**Comment by broskoski** [comment]
@dzucconi re: forms styling – https://app.zeplin.io/project/59b8592b1f1af27f2383947d/screen/5a8b51a1a4eff3566aef9013

**Comment by dzucconi** [comment]
Just a note @broskoski that I've started to work on this so confer with me if you are going to make any further edits before I push.

**Pull request closed without merging**